src/library/trackset/crate/cratefeature.cpp

```diff
@@ -451,9 +450,7 @@ void CrateFeature::slotDuplicateCrate() {
         CrateId crateId =
```

Comment:
duplicateCrate() may return an invalid id if the user aborted the action. In this case the last crate should remain activated.

Reply:
activatCrate() has it's own check, thus I removed it here. but I don't understand why double-check because failure has no consequences inside this method.
https://github.com/mixxxdj/mixxx/blob/0d58e4ee2a2dc7ae3626fda5e84b07517d57c5c6/src/library/trackset/crate/cratefeature.cpp#L294-L298

Unrelated, yes, so I'll revert.

Reply:
The check in `CrateFeature::activateCrate()` is a debug assertion that we don't want to hit.

I am not an advocate of VERIFY_OR_DEBUG_ASSERT, which weakens the purpose of debug assertions. Even if the precondition is checked twice it is still an assertion. If `CrateFeature::activateCrate()` should accept invalid crate ids then we need to replace this assertion with a regular conditional.